app/tests/retina_importers_tests/helpers.py

```diff
@@ -0,0 +1,207 @@
+    files = create_test_images()
+    data = read_json_file("upload_image_valid_data.json")
+    # create request payload
+    data.update({"image_hd": files["mhd"], "image_raw": files["zraw"]})
+    return data
+
+
+def create_upload_image_invalid_test_data():
+    # create image
+    files = create_test_images()
+    data = read_json_file("upload_image_invalid_data.json")
+    # create request payload
+    data.update({"image_hd": files["mhd"], "image_raw": files["zraw"]})
+    return data
+
+
+def remove_test_image(response):
+    # Remove uploaded test image from filesystem
+    response_obj = json.loads(response.content)
+    full_path_to_image = settings.APPS_DIR / Path(
+        response_obj["image"]["image"][1:]
+    )
+    Path.unlink(full_path_to_image)
+
+
+def get_response_status(
+    client, reverse_name, data, user="anonymous", annotation_data=None
+):
+    auth_header = get_auth_token_header(user)
+    url = reverse(reverse_name)
+    if annotation_data:
+        # create objects that need to exist in database before request is made
+        patient = PatientFactory(name=data.get("patient_identifier"))
+        existing_models = {"studies": [], "series": [], "images": []}
+        images = []
+        for data_row in data.get("data"):
+            if (
+                data_row.get("study_identifier")
+                not in existing_models["studies"]
+            ):
+                study = StudyFactory(
+                    name=data_row.get("study_identifier"), patient=patient
+                )
+                existing_models["studies"].append(study.name)
+            else:
+                study = Study.objects.get(
+                    name=data_row.get("study_identifier")
+                )
+
+            if (
+                data_row.get("image_identifier")
+                not in existing_models["images"]
+            ):
+                image = ImageFactory(
+                    name=data_row.get("image_identifier"), study=study
+                )
+                existing_models["images"].append(image.name)
+                images.append(image)
+        archive = ArchiveFactory(
+            name=data.get("archive_identifier"), images=images
+        )
+
+        response = client.post(
+            url,
+            data=json.dumps(data),
+            content_type="application/json",
+            **auth_header,
+        )
+    else:
+        response = client.post(url, data=data, **auth_header)
+    return response.status_code
+
+
+def create_test_method(
+    reverse_name, data, user, expected_status, annotation_data=None
+):
+    def test_method(self, client):
+        response_status = get_response_status(
+            client, reverse_name, data, user, annotation_data
+        )
+        assert response_status == expected_status
+
+    return test_method
+
+
+def batch_test_upload_views(batch_test_data, test_class):
+    for name, test_data in batch_test_data.items():
+        user_status_tuple = (
```

Comment:
Use pytest parametrize

Reply:
I will gradually refactor this when I need to make changes to the tests. See #677